bitcash/network/APIs/BitcoinDotComAPI.py

```diff
@@ -0,0 +1,132 @@
+                    "https://rest.bitcoin.com/v2/"],
+        "testnet": ["https://trest.bitcoin.com/v2/"],
+        "regtest": ["http://localhost:12500/v2/"],
+    }
+
+    # Paths specific to rest.bitcoin.com-based endpoints
+    PATHS = {
+        "unspent": "address/utxo/{}",
+        "address": "address/details/{}",
+        "raw-tx": "rawtransactions/sendRawTransaction/{}",
+        "tx-details": "transaction/details/{}",
+    }
+
+    @classmethod
+    def get_default_endpoints(cls, network):
+        return cls.DEFAULT_ENDPOINTS[network]
+
+    def make_endpoint_url(self, path):
+        return self.network_endpoint + self.PATHS[path]
+
+    def get_balance(self, address, *args, **kwargs):
+        api_url = self.make_endpoint_url("address").format(address)
+        r = requests.get(api_url, *args, **kwargs)
+        r.raise_for_status()
+        data = r.json()
+        return data["balanceSat"] + data["unconfirmedBalanceSat"]
+
+    def get_transactions(self, address, *args, **kwargs):
+        api_url = self.make_endpoint_url("address").format(address)
+        r = requests.get(api_url, *args, **kwargs)
+        r.raise_for_status()
+        return r.json()["transactions"]
+
+    def get_transaction(self, txid, *args, **kwargs):
+        api_url = self.make_endpoint_url("tx-details").format(txid)
+        r = requests.get(api_url, *args, **kwargs)
+        r.raise_for_status()
+        response = r.json(parse_float=Decimal)
+
+        tx = Transaction(
+            response["txid"],
+            response["blockheight"],
+            (Decimal(response["valueIn"]) * BCH_TO_SAT_MULTIPLIER).normalize(),
+            (Decimal(response["valueOut"]) * BCH_TO_SAT_MULTIPLIER).normalize(),
+            (Decimal(response["fees"]) * BCH_TO_SAT_MULTIPLIER).normalize(),
+        )
+
+        for txin in response["vin"]:
+            part = TxPart(txin["cashAddress"], txin["value"], txin["scriptSig"]["asm"])
+            tx.add_input(part)
+
+        for txout in response["vout"]:
+            addr = None
+            if (
+                "cashAddrs" in txout["scriptPubKey"]
+                and txout["scriptPubKey"]["cashAddrs"] is not None
+            ):
+                addr = txout["scriptPubKey"]["cashAddrs"][0]
+
+            part = TxPart(
+                addr,
+                (Decimal(txout["value"]) * BCH_TO_SAT_MULTIPLIER).normalize(),
+                txout["scriptPubKey"]["asm"],
+            )
+            tx.add_output(part)
+
+        return tx
+
+    def get_tx_amount(self, txid, txindex, *args, **kwargs):
+        api_url = self.make_endpoint_url("tx-details").format(txid)
+        r = requests.get(api_url, *args, **kwargs)
+        r.raise_for_status()
+        response = r.json(parse_float=Decimal)
+        return (
+            Decimal(response["vout"][txindex]["value"]) * BCH_TO_SAT_MULTIPLIER
+        ).normalize()
+
+    def get_unspent(self, address, *args, **kwargs):
```

Comment:
Do you mean `get_unspents()`?

Reply:
For the sake of respecting backwards compatibility I didn't change any function names that were there previously. This is how it was written when I found it. If I changed it in the BitcoinDotCom, it would seem natural to also reflect that in the NetworkAPI class, and hence anyone's existing code thats calls `get_unspent()` would break.

Reply:
Ah, my bad. I see.